src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/DoNotDeclareStaticMembersOnGenericTypesTests.cs

```diff
@@ -12,12 +12,268 @@ public class DoNotDeclareStaticMembersOnGenericTypesTests : DiagnosticAnalyzerTe
+    );
+        }
+
+        [Fact]
+        public void Basic_CA1000_ShouldGenerate()
+        {
+            VerifyBasic(@"Imports System
+Public Class GenericType1(Of T)
+    Private Sub New()
+    End Sub
+
+    Public Shared Sub Output(data As T)
+        Console.Write(data)
+    End Sub
+
+    Public Shared ReadOnly Property Test() As String
+        Get
+            Return String.Empty
+        End Get
+    End Property
+End Class
+
+Public NotInheritable Class GenericType2(Of T)
+    Public Shared Sub Output(data As T)
+        Console.Write(data)
+    End Sub
+
+    Public Shared ReadOnly Property Test() As String
+        Get
+            Return String.Empty
+        End Get
+    End Property
+End Class",
+    GetBasicResultAt(6, 23, DoNotDeclareStaticMembersOnGenericTypesAnalyzer.RuleId, DoNotDeclareStaticMembersOnGenericTypesAnalyzer.Rule.MessageFormat.ToString()),
+    GetBasicResultAt(10, 37, DoNotDeclareStaticMembersOnGenericTypesAnalyzer.RuleId, DoNotDeclareStaticMembersOnGenericTypesAnalyzer.Rule.MessageFormat.ToString()),
+    GetBasicResultAt(18, 23, DoNotDeclareStaticMembersOnGenericTypesAnalyzer.RuleId, DoNotDeclareStaticMembersOnGenericTypesAnalyzer.Rule.MessageFormat.ToString()),
+    GetBasicResultAt(22, 37, DoNotDeclareStaticMembersOnGenericTypesAnalyzer.RuleId, DoNotDeclareStaticMembersOnGenericTypesAnalyzer.Rule.MessageFormat.ToString())
+    );
+        }
+
+        [Fact]
+        public void CSharp_CA1000_ShouldNotGenerate()
+        {
+            VerifyCSharp(@"
+using System;
+
+public class GenericType1<T>
+{
+    private GenericType1()
+    {
+    }
+ 
+    static GenericType1()
+    {
+    }
+
+    protected static string TestProtected
+    {
+        get { return string.Empty; }
+    }
+
+    protected internal static string TestProtectedInternal
+    {
+        get { return string.Empty; }
+    }
+
+    internal static string TestInternal
+    {
+        get { return string.Empty; }
+    }
+
+    private static string TestPrivate
+    {
+        get { return string.Empty; }
+    }
+
+    protected static void OutputProtected(T data)
+    {
+        Console.Write(data);
+    }
+
+    protected internal static void OutputProtectedInternal(T data)
+    {
+        Console.Write(data);
+    }
+
+    internal static void OutputInternal(T data)
+    {
+        Console.Write(data);
+    }
+
+    private static void OutputPrivate(T data)
+    {
+        Console.Write(data);
+    }
+
+    public override bool Equals(object obj)
+    {
+        return base.Equals(obj);
+    }
+
+    public override int GetHashCode()
+    {
+        return base.GetHashCode();
+    }
+
+    public static bool operator ==(GenericType1<T> left, GenericType1<T> right)
+    {
+        return object.Equals(left, right);
+    }
+
+    public static bool operator !=(GenericType1<T> left, GenericType1<T> right)
+    {
+        return !object.Equals(left, right);
+    }
```

Comment:
I will take my comment back. `==` and `!=` are a special case for the rule not to fire because there are other rules like [CA2231](https://msdn.microsoft.com/en-us/library/ms182359.aspx) and [CA1815](https://msdn.microsoft.com/en-us/library/ms182276.aspx) which asks for the operator overloads hence we are making this exception